flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -276,38 +276,7 @@ class ExprLowering {
   fir::ExtendedValue genLoad(const fir::ExtendedValue &addr) {
```

Comment:
It may be worth to add a comment that `genLoad` is a no-op on characters and arrays.